src/StoreKeeper/WooCommerce/B2C/PaymentGateway/PaymentGateway.php

```diff
@@ -280,13 +247,116 @@ public function onReturn()
         wp_redirect($url);
     }
 
+    public function createRefundPayment($orderId, $refundId)
+    {
+        $refund = wc_get_order($refundId);
+        $refundAmount = $refund->get_total();
+
+        $api = StoreKeeperApi::getApiByAuthName();
+        $paymentModule = $api->getModule('PaymentModule');
+        try {
+            if (!self::refundExists($orderId, $refundId)) {
+                $paymentId = $paymentModule->newWebPayment([
+                    'amount' => -abs($refundAmount), // Refund should be negative
+                    'description' => sprintf(
+                        __('Refund via Wordpress plugin (Refund #%s)', I18N::DOMAIN),
+                        $refundId
+                    ),
+                ]);
+                self::addRefund($orderId, $paymentId, $refundId, $refundAmount);
+
+                $orderHandler = new OrderHandler();
+                $task = $orderHandler->create($orderId);
+
+                if (!$task) {
+                    throw new \Exception('Order export task was not created');
+                }
+            }
+        } catch (\Throwable $exception) {
+            LoggerFactory::create('refund')->error($exception->getMessage(), $exception->getTrace());
```

Comment:
if this fails it would be never handled anymore, maybe it's better to add it to Refund table, and spawn a task to handle it (add refund in storekeeper) later? Than we don't loose it in case the error. You already have a 'is_synced; so maybe use that one.

Reply:
Added `scheduleRefundTask` when it failed.